### PR TITLE
🐛 fix(openapi): keep /users/me reachable by restricted API keys

### PR DESCRIPTION
**Pull request by rdmclin2**
### 💻 Change Type

- [ ] ✨ feat
- [x] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 🔨 chore

### 🔀 Description of Change

`GET /api/v1/users/me` picked up a `requireApiKeyScope('user:read')` guard in #18029. That gate strands a valid API key outside the product: `lh login` resolves its userId from this endpoint (`apps/cli/src/auth/apiKey.ts` → `getUserIdFromApiKey()`), so a key minted without `user:read` now fails to log in with a scope error.

The endpoint returns only the caller's **own** identity — their user row and their own roles; the CLI even passes `includeCount=0` to skip the message count. It therefore discloses nothing the key holder is not already, which is why GitHub's `/user` is likewise open to any token. Identity bootstrap has to sit in front of the scope system, not behind it.

Removes the gate and replaces it with a comment stating why the route is deliberately open, so the next review does not re-tighten it.

**Blast radius of the bug being fixed:** keys with `scopes IS NULL` (every key issued before scopes existed) and explicit full-access keys were unaffected — only newly created restricted keys omitting `user:read` could hit it.

### 📝 Additional Information

Regression test added: a restricted key holding only `agent:read` (and one holding only `model:invoke`) must reach `/users/me`. Verified it fails when the guard is put back.

Fixes LOBE-12934

### CODE REVIEW

**Comment by rdmclin2** [comment]
@codex review

File: packages/openapi/src/routes/users.route.ts [hunk 1]
```diff
@@ -26,8 +26,14 @@ UserRoutes.get(
   '/me',
   describeRoute({ summary: 'Get current authenticated user', tags: ['users'] }),
   requireAuth,
-  // no RBAC permission, but restricted API keys still need the user domain
-  requireApiKeyScope('user:read'),
+  // Deliberately reachable by every authenticated caller, including restricted
+  // API keys that hold no `user:read` — please do not add a scope gate here.
```
[the remark this answers is not in the document]
Comment (rdmclin2) [reply packages/openapi/src/routes/users.route.ts]: Fixed in 9c0ff71466 — you are right that `messageCount` is not identity, and my PR description overclaimed when it said the endpoint returns nothing beyond the caller's own identity.

`UserController.getCurrentUser` now computes `includeCount` as `countRequested && canReadChatCounts(c)`: a restricted key needs `chat:read` to get the count; session/OIDC callers and full-access keys are unchanged.

**Withheld rather than rejected**, deliberately: `includeCount` defaults to true, so a 403 would make identity resolution depend on a query parameter — reintroducing exactly the trap this PR removes for callers that omit `includeCount=0`. The identity payload is always returned; only the count drops out.

Covered by 4 controller tests (no chat scope → `false`, `chat:read` → `true`, full-access/`NULL`/OIDC → `true`, explicit `includeCount=0` still honoured). Verified they fail when the scope check is bypassed. Also added the `@/envs` alias to the package's vitest config, which the controller test needed.

**Comment by rdmclin2** [comment]
@codex review latest code

File: packages/openapi/src/routes/users.route.ts [hunk 2]
```diff
@@ -26,8 +26,14 @@ UserRoutes.get(
...
+  // It returns only the caller's OWN identity (their user row and their own
```
[the remark this answers is not in the document]
Comment (rdmclin2) [reply packages/openapi/src/routes/users.route.ts]: Fixed in 8d35d171c5 — agreed, and this is the version the endpoint should have shipped as: reachability and payload are two separate questions, and only the first one needed loosening.

`/me` stays open to every authenticated caller, but the response is now scoped:

- **without `user:read`** → narrowed to `{ id }`, which is all `lh login` consumes (`body.data.id`). Email, phone, preferences and role names no longer reach a key granted only `model:invoke` / `agent:read`.
- **without `chat:read`** → `messageCount` additionally withheld (previous commit).
- session / OIDC and full-access keys → unchanged full payload.

The two checks are now one helper (`callerHasScope`), so a third field-level gate is a one-liner rather than another bespoke branch.

3 more tests cover the projection (narrowed without the scope / full with `user:read` / full for `*`, `NULL` and OIDC), verified failing when the projection is removed. 9 tests total on this PR.

**Comment by rdmclin2** [comment]
@codex review latest code

File: packages/openapi/src/routes/users.route.ts [hunk 3]
```diff
@@ -26,8 +26,14 @@ UserRoutes.get(
...
+  // roles), so it discloses nothing the key holder is not already, and it is
```
[the remark this answers is not in the document]
Comment (rdmclin2) [reply packages/openapi/src/routes/users.route.ts]: Fixed in d6f3a83a3d. The comment was written for the pre-projection version and, since it exists precisely to stop someone re-gating the route, a stale description of the contract was the worst thing to leave behind.

It now separates the two halves explicitly: the route stays reachable for identity bootstrap, and **reachability is not disclosure** — the payload is scoped inside the controller (`{ id }` without `user:read`, `messageCount` needs `chat:read`), with a pointer to widen it there under a scope check rather than by gating the route.

**Comment by rdmclin2** [comment]
@codex review latest code
